[Tune] Add ZOOpt search algorithm (#7960)
ray-project/ray: doc/source/tune-searchalg.rst

    :show-inheritance:
    :noindex:

ZOOpt Search
------------

> The ``ZOOptSearch`` is a list of derivative-free optimization algorithms that are backed by `ZOOpt <https://github.com/polixir/ZOOpt>`__ package to perform sequential model-based hyperparameter optimization. Currently, Asynchronous Sequential RAndomized COordinate Shrinking (ASRacos) algorithm is implemented in Tune. Note that this class does not extend ``ray.tune.suggest.BasicVariantGenerator``, so you will not be able to use Tune’s default variant generation/search space declaration when using ZOOptSearch.

Reviewer: ```suggestion
The ``ZOOptSearch`` is a SearchAlgorithm for derivative-free optimization. It is backed by the `ZOOpt <https://github.com/polixir/ZOOpt>`__ package. Currently, Asynchronous Sequential RAndomized COordinate Shrinking (ASRacos) algorithm is implemented in Tune. Note that this class does not extend ``ray.tune.suggest.BasicVariantGenerator``, so you will not be able to use Tune’s default variant generation/search space declaration when using ZOOptSearch.
```
